Ubuntu-Mate (16.04.3) Source Compile error

Raspberry Pi 3 (Turtlebot3) w/V2 camera, Linux cassini 4.9.53-v7+ #1040 SMP Fri Oct 6 14:19:18 BST 2017 armv7l armv7l armv7l GNU/Linux, 
following https://github.com/UbiquityRobotics/raspicam_node/issues/new instructions.

Having to reinstall Ubuntu-Mate/ROS and raspicam_node (due SD Card crash) in my RasPi3, resulting in many compile errors in the "RaspiCamControl.h" . Thus the "make -j4 -l4" failed. Appreciate your advice. Ross

[ 84%] Built target turtlebot3_msgs_generate_messages
[ 86%] Generating dynamic reconfigure files from cfg/Camera.cfg: /home/turtlebot/catkin_ws/devel/include/raspicam_node/CameraConfig.h /home/turtlebot/catkin_ws/devel/lib/python2.7/dist-packages/raspicam_node/cfg/CameraConfig.py
Scanning dependencies of target raspicli
Scanning dependencies of target raspicamcontrol
[ 88%] Building CXX object raspicam_node/CMakeFiles/raspicli.dir/src/RaspiCLI.cpp.o
[ 90%] Building CXX object raspicam_node/CMakeFiles/raspicamcontrol.dir/src/RaspiCamControl.cpp.o
Generating reconfiguration files for Camera in raspicam_node
Wrote header file in /home/turtlebot/catkin_ws/devel/include/raspicam_node/CameraConfig.h
[ 90%] Built target raspicam_node_gencfg
In file included from /home/turtlebot/catkin_ws/src/raspicam_node/src/RaspiCamControl.cpp:37:0:
/home/turtlebot/catkin_ws/src/raspicam_node/include/RaspiCamControl.h:116:5: error: ‘MMAL_PARAM_EXPOSUREMODE_T’ does not name a type
     MMAL_PARAM_EXPOSUREMODE_T exposureMode;
     ^
/home/turtlebot/catkin_ws/src/raspicam_node/include/RaspiCamControl.h:117:5: error: ‘MMAL_PARAM_EXPOSUREMETERINGMODE_T’ does not name a type
     MMAL_PARAM_EXPOSUREMETERINGMODE_T exposureMeterMode;. 

and many more similar errors . The RaspiCamControl,h file is in the include directory

--- a/include/RaspiCamControl.h
+++ b/include/RaspiCamControl.h
@@ -29,6 +29,13 @@ SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.
 #ifndef RASPICAMCONTROL_H_
 #define RASPICAMCONTROL_H_
 
+#include "interface/mmal/mmal.h"
+#include "interface/mmal/mmal_logging.h"
+#include "interface/mmal/util/mmal_default_components.h"
+#include "interface/mmal/util/mmal_util.h"
+#include "interface/mmal/util/mmal_util_params.h"
+#include "interface/vmcs_host/vc_vchi_gencmd.h"
+
 /* Various parameters
  *
  * Exposure Mode

